File: 2024/2024-11-29/fizzbuzz.rb

def fizzbuzz n
    p case
    when n % 15 == 0
        'fizzbuzz'
    when n % 5 == 0
        'buzz'
    when n % 3 == 0
        'fizz'
    else
        n.to_s
    end
end

def run
    (1..30).map { |n| fizzbuzz n }
end

if __FILE__ == $0
    run
end

File: 2025/2025-01-24/fizzbuzz.ts
function fizzbuzz(n: number): string {
    let result = n.toString();
    if (n % 15 == 0) {
        result = 'fizzbuzz';
    } else if (n % 5 == 0) {
        result = 'buzz';
    } else if (n % 3 == 0) {
        result = 'fizz';
    }
    return result
}

function run() {
    for (let i = 1; i <= 100; i++) {
        console.log(fizzbuzz(i))
    }
}

if (require.main === module) {
    run();
}